windows.Dockerfile

```diff
@@ -0,0 +1,99 @@
+
+# ------------------------------------------------------------------------
+
+FROM mcr.microsoft.com/dotnet/framework/sdk:4.8
+
+SHELL ["powershell.exe", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+# See https://github.com/jenkinsci/docker-agent/blob/master/8/windows/windowsservercore-1809/Dockerfile
+
+ENV JAVA_VERSION jdk8u252-b09
+
+USER ContainerAdministrator
+
+RUN Write-Host ('Downloading https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u252-b09.1/OpenJDK8U-jdk_x64_windows_hotspot_8u252b09.zip ...'); `
+    [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12; `
+    Invoke-WebRequest -Uri https://github.com/AdoptOpenJDK/openjdk8-binaries/releases/download/jdk8u252-b09.1/OpenJDK8U-jdk_x64_windows_hotspot_8u252b09.zip -O 'openjdk.zip'; `
+    ls; `
+    Write-Host ('Verifying sha256 (4e2c92ba17481321eaeb1769e85eec99a774102eb80b700a201b54b130ab2768) ...'); `
+    if ((Get-FileHash openjdk.zip -Algorithm sha256).Hash -ne '4e2c92ba17481321eaeb1769e85eec99a774102eb80b700a201b54b130ab2768') { `
+            Write-Host 'FAILED!'; `
+            exit 1; `
+    }; `
+    `
+    Write-Host 'Expanding Zip ...'; `
+    Expand-Archive -Path openjdk.zip -DestinationPath C:\ ; `
+    $jdkDirectory=(Get-ChildItem -Directory | ForEach-Object { $_.FullName } | Select-String 'jdk'); `
+    Move-Item -Path $jdkDirectory C:\openjdk-8; `
+    Write-Host 'Removing openjdk.zip ...'; `
+    Remove-Item openjdk.zip -Force
+
+ARG GIT_VERSION=2.26.0
+ARG GIT_PATCH_VERSION=1
+RUN [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12 ; `
+    $url = $('https://github.com/git-for-windows/git/releases/download/v{0}.windows.{1}/MinGit-{0}-64-bit.zip' -f $env:GIT_VERSION, $env:GIT_PATCH_VERSION) ; `
+    Write-Host "Retrieving $url..." ; `
+    Invoke-WebRequest $url -OutFile 'mingit.zip' -UseBasicParsing ; `
+    Expand-Archive mingit.zip -DestinationPath c:\mingit ; `
+    Remove-Item mingit.zip -Force
+
+ARG GIT_LFS_VERSION=2.10.0
+RUN [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12 ; `
+    $url = $('https://github.com/git-lfs/git-lfs/releases/download/v{0}/git-lfs-windows-amd64-v{0}.zip' -f $env:GIT_LFS_VERSION) ; `
+    Write-Host "Retrieving $url..." ; `
+    Invoke-WebRequest $url -OutFile 'GitLfs.zip' -UseBasicParsing ; `
+    Expand-Archive GitLfs.zip -DestinationPath c:\mingit\mingw64\bin ; `
+    Remove-Item GitLfs.zip -Force ; `
+    & C:\mingit\cmd\git.exe lfs install ; `
+    $CurrentPath = (Get-Itemproperty -path 'hklm:\system\currentcontrolset\control\session manager\environment' -Name Path).Path ; `
+    $NewPath = $CurrentPath + ';C:\mingit\cmd;C:\openjdk-8\bin' ; `
+    Set-ItemProperty -path 'hklm:\system\currentcontrolset\control\session manager\environment' -Name Path -Value $NewPath
+
+ARG user=jenkins
+
+ARG AGENT_FILENAME=agent.jar
+ARG AGENT_HASH_FILENAME=$AGENT_FILENAME.sha1
+
```

Comment:
@gayashanbc better to add comments for each `RUN` instruction indicating the task carried out.

Reply:
Resolved in https://github.com/wso2/build-docker/pull/18/commits/8bd6ae87c078b011e391265c6f7a479d59c8b05e